File: BackEnd/gamedex/src/main/java/cat/copernic/gamedex/entity/LoginResponse.java
package cat.copernic.gamedex.entity;

import org.springframework.data.mongodb.core.mapping.Document;
import lombok.Data;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

/**
 * Classe que representa una resposta d'inici de sessió.
 */
@Document(collection = "loginResponse")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class LoginResponse {

    /**
     * El nom d'usuari de la resposta d'inici de sessió.
     */
    private String username;

    /**
     * El tipus d'usuari de la resposta d'inici de sessió.
     */
    private UserType userType;
}

File: FrontEnd/app/src/main/java/cat/copernic/grup4/gamedex/Core/Model/LoginResponse.kt
package cat.copernic.grup4.gamedex.Core.Model

/**
 * Classe que representa una resposta d'inici de sessió.
 *
 * @param username El nom d'usuari de la resposta d'inici de sessió.
 * @param userType El tipus d'usuari de la resposta d'inici de sessió.
 */
data class LoginResponse(
    val username: String,
    val userType: UserType
)